modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/DXPActivationKeysTable/Header/index.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayAlert from '@clayui/alert';
+
+import classNames from 'classnames';
+import {useMemo, useState} from 'react';
+import {Button, ButtonDropDown} from '../../../../../common/components';
+import {ALERT_DOWNLOAD_TYPE} from '../../../utils/constants/alertDownloadType';
+import {AUTO_CLOSE_DOWNLOAD_ALERT_TIME} from '../../../utils/constants/autoCloseDownloadAlertTime';
+import {DOWNLOADABLE_LICENSE_KEYS} from '../utils/constants';
+import {ALERT_ACTIVATION_AGGREGATED_KEYS_DOWNLOAD_TEXT} from '../utils/constants/alertAggregateKeysDownloadText';
+import {getActivationKeyDownload} from '../utils/getActivationKeyDownload';
+import {getActivationKeysActionsItems} from '../utils/getActivationKeysActionsItems';
+import {getActivationKeysDownloadItems} from '../utils/getActivationKeysDownloadItems';
+
+const DXPActivationKeysTableHeader = ({
+	accountKey,
+	activationKeys,
+	licenseKeyDownloadURL,
+	selectedKeys,
+	sessionId,
+}) => {
+	const [
+		activationKeysDownloadStatus,
+		setActivationKeysDownloadStatus,
+	] = useState('');
+
+	const selectedKeysIDs = selectedKeys
+		.map((selectedKey) => `licenseKeyIds=${selectedKey}`)
+		.join('&');
```

Comment:
.map + .join do 2 iterations, if use reduce will do only one iteration. It's possible we measure the different performance?

Reply:
I don't know for sure, we need to be sure that the & character does not fill the last index, because we will finish like this
`licenseKeyIds=1&licenseKeyIds=2&`
The join grant us that this will never happen, and we do not need to do any other validation

Reply:
Yes, make senses! Thank you